app/views/tracks.js

```diff
@@ -11,6 +11,92 @@ const TracksView = Mn.CompositeView.extend({
 
     childView: TrackView,
 
+    initialize() {
+        this.dragObj = {
+            'dragDown': false,
+            'model': undefined,
+            'newIndex': undefined
+        };
+        $(document).mousemove((e) => {
+            if (this.dragObj.dragDown) {
+                this.drag(e);
+            }
+        });
+        $(document).mouseup((e) => {
+            if (!this.dragObj.dragDown) return;
+
+            // Reset all Style
+            this.$el.find('#track-list').children()
+                .removeClass('dragged dragover above below');
+
+            // Reorder the track to the new position
+            let tracks = application.appState.get('currentPlaylist');
+            tracks.reorderTrack(this.dragObj.model, this.dragObj.newIndex);
+
+            // Reset the drag object
+            this.dragObj = {
+                'dragDown': false,
+                'model': undefined,
+                'newIndex': undefined
+            };
+        });
+        application.channel.reply('track:drag', (track) => {
+            this.dragObj.dragDown = true;
+            this.dragObj.model = track;
+        }, this);
+    },
```

Comment:
Could you refactor this part, I fear this will become hard to read later. I suggest:
- a `<prototype>.bindDragEvents` method that bind `mousemove` and `mouseup` from `document
- a `<prototype>.dragStart` for the `mousedown .drag` event handler (see comment in `track` view below)
- a `<protoype>.drag` that enable draggin (with a early return if `this.dragObj.dragDown` is falsy) bound to `mousemove`
- a `<prototype>.dragEnd` that execute drag results parsing, bound to `mouseup`

What do you think about it?